enterprise/server/workflow/config/config.go

```diff
@@ -238,15 +238,26 @@ func prepareKytheOutputs(dirName string) string {
 	buf := fmt.Sprintf(`
 export KYTHE_DIR="$BUILDBUDDY_CI_RUNNER_ROOT_DIR"/%s
 ulimit -n 10240
-find -L bazel-out/ -name *.go.kzip -print0 | xargs -r0 zipmerge output.go.kzip
-find -L bazel-out/ -name *.protobuf.kzip -print0 | xargs -r0 zipmerge output.protobuf.kzip
 
-if [ -f output.go.kzip ]; then
-  "$KYTHE_DIR"/indexers/go_indexer -continue output.go.kzip >> kythe_entries
-fi
-if [ -f output.protobuf.kzip ]; then
-  "$KYTHE_DIR"/indexers/proto_indexer -index_file output.protobuf.kzip >> kythe_entries
-fi
+find -L ../bazel-out/ -name "*.go.kzip" | parallel --gnu $KYTHE_DIR/indexers/go_indexer -continue | $KYTHE_DIR/tools/dedup_stream >> kythe_entries
+find -L ../bazel-out/ -name "*.proto.kzip" | parallel --gnu $KYTHE_DIR/indexers/proto_indexer -index_file {} | $KYTHE_DIR/tools/dedup_stream >> kythe_entries
+find -L bazel-out -name '*.java.kzip' | parallel --gnu  java -jar $KYTHE_DIR/indexers/java_indexer.jar | $KYTHE_DIR/tools/dedup_stream >> kythe_entries
+
+# cxx indexing needs a cache to complete in a "reasonable" amount of time. It still takes a long time
+# and produces very large indices.
+# See https://groups.google.com/g/kythe/c/xKXE3S1JIRI for discussion of these args.
+# TODO(jdelfino): Will this work? Is there a better way? Does this workflow need its own docker image?
+sudo apt update && sudo apt install -y memcached
```

Comment:
maybe we should put this behind an if check so we don't run it if memcached exists? Apt is slow

Reply:
+1 though IDK if we have recycling enabled for these workflows?

also @jdelfino re

> I'm guessing this will run into a permissions issue.

We have passwordless sudo enabled, so it should work

Reply:
Also, we can skip memcached entirely if there are no C++ files... I included that case in the check